py/torch_tensorrt/dynamo/conversion/impl/select.py

```diff
@@ -61,3 +69,271 @@ def select(
+        # x[:,ind_1, :, ind_2] = 2, 3, 4, 10, 30, 50
+        transpose_layer = network.add_shuffle(input)
+        new_order = []
+        for i in range(adv_indx_count):
+            new_order.append(adv_indx_indices[i])
+        for i in range(rank):
+            if i not in adv_indx_indices:
+                new_order.append(i)
+        _LOGGER.debug(f"The new transpose order is {new_order}")
+        transpose_layer.second_transpose = tuple(new_order)
+        set_layer_name(transpose_layer, target, name + "_index_transpose", source_ir)
+        transpose_tensor = transpose_layer.get_output(0)
+
+        # Flatten [x_1, x_2,.......x_m, y_1, y_2,.....y_n]
+        # transpose_tensor_shape = network.add_shape(transpose_tensor)
+        transpose_tensor_shape = transpose_tensor.shape
+        _LOGGER.debug(f"The shape of transpose tensor is {transpose_tensor_shape}")
+        mult_d0 = 1
+        for i in range(adv_indx_count):
+            mult_d0 = mult_d0 * transpose_tensor_shape[i]
+        mult_d1 = 1
+        for i in range(adv_indx_count, rank):
+            mult_d1 = mult_d1 * transpose_tensor_shape[i]
+
+        concat_tensor_layer = network.add_concatenation(
+            [
+                get_trt_tensor(network, mult_d0, "d0_shape"),
+                get_trt_tensor(network, mult_d1, "d1_shape"),
+            ]
+        )
+        set_layer_name(concat_tensor_layer, target, name + "_index_Concat", source_ir)
+        concat_tensor = concat_tensor_layer.get_output(0)
+
+        reshape_layer = network.add_shuffle(transpose_tensor)
+        # check this
+        reshape_layer.set_input(1, concat_tensor)
+        flatten_tensor = reshape_layer.get_output(0)
+        _LOGGER.debug(f"The flatten tensor shape is {flatten_tensor.shape}")
+
+        # tensor index = \sum_{i=1}^m (ind_i * \prod_{j=i+1}^m (x_j)),  ind_i is input indices[i], x_j is the
+        # // j dimension of input x.
+        multiplier = get_trt_tensor(
+            network, dim_tensor_list[adv_indx_indices[adv_indx_count - 1]], "dim_last"
+        )
+        cum_adv_index = tensor_indices[adv_indx_count - 1]
+        for i in range(adv_indx_count - 2, -1, -1):
+            adv_index = convert_binary_elementwise(
+                network,
+                target,
+                source_ir,
+                name + "index_intermediate",
+                trt.ElementWiseOperation.PROD,
+                multiplier,
+                tensor_indices[i],
+            )
+            cum_adv_index = convert_binary_elementwise(
+                network,
+                target,
+                source_ir,
+                name + "index_sum_intermediate",
+                trt.ElementWiseOperation.SUM,
+                cum_adv_index,
+                adv_index,
+            )
+            multiplier = convert_binary_elementwise(
+                network,
+                target,
+                source_ir,
+                name + "index_intermediate",
```

Comment:
Needs to be `_index_prod_intermediate_{i}` to avoid clashing with `adv_index`, which has the same layer name